clients/drcachesim/analyzer_multi.cpp

```diff
@@ -139,8 +139,7 @@ analyzer_multi_t::create_invariant_checker()
         // reader(s) for seeking. For now we only read them for this test.
         // TODO i#5843: Share this code with scheduler_t or pass in for all
         // tools from here for fast skipping in serial and per-cpu modes.
-        std::string tracedir =
-            raw2trace_directory_t::tracedir_from_rawdir(op_indir.get_value());
+        std::string tracedir = get_input_dir();
```

Comment:
tracedir_from_rawdir seems to handle certain cases that get_input_dir doesn't. E.g., if op_indir is specified, it is returned directly, whereas previously tracedir_from_rawdir would ensure it ends in '/trace'. Is this a bug?

Reply:
-indir supports any trace dir: it doesn't have to be structured as a subdirectory trace/ with a peer raw/.  We want flexibility there.  The initial layout is created by us but a postprocessed trace dir can be copied somewhere else and still work with -indir.  Added a comment.

Reply:
> We want flexibility there

Based on the comments in tracedir_from_rawdir, isn't it already flexible?

```
// If it ends in "/raw" or a window subdir, replace with "/trace".
..
// If it contains a "/raw" or "/trace" subdir, add "/trace" to it.
..
// Use it directly.
```

Further, today we can specify `<dest-dir>` or `<dest-dir>/raw` and expect it to just work. Is it still the case with this change in create_invariant_checker?

Reply:
The logic below would seem to just disable the serial_schedule_file and cpu_schedule_file checks if they are not found, with no way to know there was an error in finding those files.

Reply:
This is inside create_invariant_checker() so only affects finding the schedule files.  The existing code has to be changed to support multi.  I made it match get_aux_file_path in using get_input_dir: but since it doesn't use the rest of get_aux_file_path (today requires a tied option; could maybe be made to use it) -- maybe the best is this which I now changed it to:

```
raw2trace_directory_t::tracedir_from_rawdir(get_input_dir());
```